packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/traits/test_data_query_types.rs

```diff
@@ -0,0 +1,70 @@
+use itertools::Itertools;
+
+use crate::{components::ComponentVariantPropTypes, props::PropValueType};
+
+use super::TryFromDataQueryResults;
+
+/// This trait implements functions needed to test the types of a `DataQueryResults` object.
+/// It **should not be depended upon at runtime**, and is implemented only for testing type consistency.
+pub trait TestDataQueryTypes: TryFromDataQueryResults {
+    /// A list of tuples `(field_name, expected_type)` that are derived from the `RequiredData` struct's fields.
+    /// For example
+    /// ```ignore
+    /// struct RequiredData {
+    ///   a: PropView<prop_type::String>,
+    ///   b: PropView<prop_type::Number>,
+    /// }
+    /// ```
+    /// would result in the list `&[("a", PropValueType::String), ("b", PropValueType::Number)]`.
```

Comment:
I believe it should be `&[("a", Some(PropValueType::String)), ("b", Some(PropValueType::Number))]`